.github/workflows/docker-v2-release.yml

```diff
@@ -0,0 +1,277 @@
+name: Docker v2 release
```

Comment:
There seems to be a lot of duplication between this and `docker-v2.yml` - possibly when we finish other clear-up, it might be worth considering if we can merge them? (possibly using `    if: github.event_name == 'release'` to replace internal publishes with publishes to Dockerhub?)